### ISSUE TITLE
Unable to compile for / install on Fedora 41

### ISSUE THREAD

**Issue by mbert** [trunkmaster/nextspace#446]
I am aware that what I am reporting here is probably only the tip of the iceberg. New compilers and new defaults keep breaking stuff that was buildable before. But here we go.

```
cd Packaging/Sources/
sh 0_build_libdispatch.sh
[...]
[ 13%] Building C object src/CMakeFiles/dispatch.dir/apply.c.o
/home/mdietze/src/build/nextspace/Packaging/Sources/BUILD_ROOT/swift-corelibs-libdispatch-swift-5.9.2-RELEASE/src/apply.c:366:4: error: cast from 'dispatch_function_t' (aka 'void (*)(void * _Nullable)') to 'dispatch_apply_function_t' (aka 'void (*)(void *, unsigned long)') converts to incompatible function type [-Werror,-Wcast-function-type-mismatch]
  366 |                         (dispatch_apply_function_t)_dispatch_Block_invoke(work));
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
gmake[2]: *** [src/CMakeFiles/dispatch.dir/build.make:90: src/CMakeFiles/dispatch.dir/apply.c.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:205: src/CMakeFiles/dispatch.dir/all] Error 2
gmake: *** [Makefile:136: all] Error 2
[ 10%] Built target BlocksRuntime
[ 13%] Building C object src/CMakeFiles/dispatch.dir/apply.c.o
/home/mdietze/src/build/nextspace/Packaging/Sources/BUILD_ROOT/swift-corelibs-libdispatch-swift-5.9.2-RELEASE/src/apply.c:366:4: error: cast from 'dispatch_function_t' (aka 'void (*)(void * _Nullable)') to 'dispatch_apply_function_t' (aka 'void (*)(void *, unsigned long)') converts to incompatible function type [-Werror,-Wcast-function-type-mismatch]
  366 |                         (dispatch_apply_function_t)_dispatch_Block_invoke(work));
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
gmake[2]: *** [src/CMakeFiles/dispatch.dir/build.make:90: src/CMakeFiles/dispatch.dir/apply.c.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:205: src/CMakeFiles/dispatch.dir/all] Error 2
gmake: *** [Makefile:136: all] Error 2

```

Since the build already fails at the first step I can't make a statement about subsequent steps. But I'd assume that things will not look any better.

To make things worse: Other than on Fedora 40 I can't even install the binary distribution on Fedora 41, because a number of dependencies are no longer available. 

I understand that this is potentially a lot of unpleasant work, and it will keep repeating with each distro upgrade. Any plans to support Fedora 41?

**Comment by armm77** [comment]
hi @mbert , before investing time in research, the most logical thing is to wait for @trunkmaster to decide to change or upload the version of the libdispatch library, since in Fedora 41 there is version 6.0.2 and this would generate confusion with the version of the project.

**Comment by mbert** [comment]
Yes, sure. Just wanted to raise awareness. On Fc40 the binary distribution for Fc39 works fine. But on Fc41 there's no way ATM. And I know from experience having to compile the mainstream GNUstep libraries on Fc41 that there's lots of issues with the compilers shipping with it.

**Comment by trunkmaster** [comment]
@mbert you don't need to build libdispatch on Fedora - just install it with `dnf install libdispatch-devel` and run next step `./1_build_libcorefoundation.sh`.

**Comment by mbert** [comment]
Thank you. However there seem to be other problems on a system where nextspace has never been installed: 

- /Develop/Makefiles/GNUstep.sh does not exist in this location with a vanilla GS installation (can be symlinked though)
- nextspace-gnustep-devel is required by nextspace-frameworks, but it is not built at this point

**Comment by trunkmaster** [comment]
> Thank you. However there seem to be other problems on a system where nextspace has never been installed:
> 
>     * /Develop/Makefiles/GNUstep.sh does not exist in this location with a vanilla GS installation (can be symlinked though)

You can't mix vanilla GS and NextSpace setup - filesystem layout differs.

>     * nextspace-gnustep-devel is required by nextspace-frameworks, but it is not built at this point

You need to install nextspace-gnustep first (steps 3-7 in Packaging/Sources).

**Comment by trunkmaster** [comment]
@mbert do you have progress or is it sill an issue to you?

**Comment by mbert** [comment]
I did run into some more issues, and since I was quite busy I did not pursue it any further at that time and decided to wait for a future release instead (in particular as with the keyboard assignment problems from #445 I cannot really use nextspace as my daily driver). At home I still have an older Fedora release on which the current version runs fine.

**Comment by trunkmaster** [comment]
Closing this issue as fixed. Please reopen if you'll face it again.

**Comment by mbert** [comment]
Thank you for your support. Just out of interest: Will there be a release (maybe even with FC41 packages) in the near future? Maybe even with a solution to #445? That would be wonderful!

**Comment by trunkmaster** [comment]
@mbert I plan to make a release after current issues will be resolved. They are:
- screen/display management (##451 
- all locale/keyboard issues.

I guess it will be a good reason for a bugfix release.
